cmd/kube-apiserver/app/server.go

```diff
@@ -63,7 +64,12 @@ func init() {
 
 // NewAPIServerCommand creates a *cobra.Command object with default parameters
 func NewAPIServerCommand() *cobra.Command {
-	s := options.NewServerRunOptions()
+	featureGate := utilfeature.DefaultMutableFeatureGate
+	featureGate.DeferErrorsToValidation(true)
+	effectiveVersion := utilversion.DefaultEffectiveVersionRegistry.EffectiveVersionForOrRegister(
+		utilversion.ComponentGenericAPIServer, utilversion.K8sDefaultEffectiveVersion())
```

Comment:
We expect this version to directly match our versions for our featuregates, which means it must be 1.31, 1.32, etc, right?  but near as i can tell `K8sDefaultEffectiveVersion` returns the component-base version that is injected at build time.